app/pages/index.tsx

```diff
@@ -29,4 +33,12 @@ const Home: NextPage = () => {
   )
 }
 
+export const getServerSideProps: GetServerSideProps = async ({ locale }) => {
+  return {
```

Comment:
Probably scope creep or better suited for a follow-up PR, but this feels like an area that could benefit from some more thinking about what pattern is best for larger codebases and long term maintainability. For instance, should this be a reusable method to support something like:

```ts
export const getServerSideProps: GetServerSideProps = async ({ locale }) => {
  return { props: setupServerI18nProps(locale) }
}
```

Something like that could also allow pages to not have a direct dependency on `next-i18next`, making it easier to swap out in the future if necessary. That method could also be a good spot for more context/documentation via a code comment – I looked at `serverSideTranslations` documentation briefly and still don't really understand what it's doing 😅